src/users/views.py

```diff
@@ -45,7 +46,11 @@ def check_authentication(self, request):
     def delete(self, request, *args, **kwargs):
         """Logout on delete."""
         self.check_authentication(request)
-        dj_rest_auth.views.LogoutView().logout(request)
+        # only delete the auth token if it was used for authentication
+        if request.auth is not None:
+            request.auth.delete()
+        # issue a django logout too
```

Comment:
could we perhaps change this to reference the fact that we are using this to log the session out?
e.g.: "use django logout to log out the active session (if any)"